logbook-core/src/main/java/org/zalando/logbook/DefaultLogbook.java

```diff
@@ -56,7 +56,7 @@
                 final Correlation<HttpRequest, HttpResponse> correlation =
                         new SimpleCorrelation<>(correlationId, duration, request, response);
                 final String message = formatter.format(correlation);
-                writer.writeResponse(new SimpleCorrelation<>(correlationId, duration, format, message));
```

Comment:
Would just different correlation type serve the need?
Lets say you introduce DelayedCorrelation (or extend simple correlation) which holds both request and response and DelayedLogWriter which does nothing on writeRequest but writes both on writeResponse thus you do not need to change `HttpLogWriter` interface